src/pspm_con1.m

```diff
@@ -1,176 +1,176 @@
+end
 for c=1:numel(convec)
   if ~isnumeric(convec{c})
-    errmsg=sprintf('Contrast #%d is not a numeric vector.', c); warning(errmsg); return;
-  end;
-end;
-
-% check contrast --
-if numel(connames)~=numel(convec)
+    warning('Contrast #%d is not a numeric vector.', c); return
+  end
+end
+% 2.4 check contrast
+if numel(connames) ~= numel(convec)
   warning('Number of contrast names (%d) and number of contrast vectors (%d) don''t match.', ...
     numel(connames), numel(convec));
-  return;
-end;
-
-% store for output
+  return
+end
+% 2.5 store for output
 out_datatype = datatype;
 switch datatype
   case 'param'
     datatype = 'stats';
   case {'cond', 'recon'}
   otherwise
     warning('Unknown datatype');
-    return;
-end;
-
-% set load1_options
+    return
+end
+% 2.6 set load1_options
 load1_options = struct('zscored',0);
 if isfield(options, 'zscored') && options.zscored
   load1_options.zscored = 1;
-end;
-
-% work on contrasts
-% ------------------------------------------------------------------------
-for iFn =1:numel(modelfile)
-  % user output --
+end
+%% 3 work on contrasts
+for iFn = 1:numel(modelfile)
+  % 3.1 user output --
   fprintf('Loading data ... ');
-
-  % retrieve stats --
-  [sts, data, mdltype] = pspm_load1(modelfile{iFn}, datatype, '', load1_options);
-  if sts == -1
+  % 3.2 retrieve stats --
+  [lsts, data, ~] = pspm_load1(modelfile{iFn}, datatype, '', load1_options);
+  if lsts == -1
     warning('ID:invalid_input', 'Could not retrieve stats');
     return;
-  end;
-  % create con structure or retrieve existing contrasts --
+  end
+  % 3.3 create con structure or retrieve existing contrasts --
   if deletecon == 1
     con = []; conno = 0;
   else
-    [sts, con] = pspm_load1(modelfile{iFn}, 'con');
-    if sts == -1
+    [lsts, con] = pspm_load1(modelfile{iFn}, 'con');
+    if lsts == -1
       fprintf(' Creating fresh contrast structure.\n');
       con = []; conno = 0;
     else
       conno = numel(con);
-    end;
-  end;
-
-  % user output --
+    end
+  end
+  % 3.4 user output --
   fprintf('\nWriting contrasts to %s\n', modelfile{iFn});
-  % check number of contrast weights --
+  % 3.5 check number of contrast weights --
   paramno = size(data.stats, 1);
-  for c=1:numel(convec)
+  for c = 1:numel(convec)
     if numel(convec{c}) > paramno
       warning('Contrast (%d) has more weights than statistics (%d) in modelfile %s', ...
-        numel(convec{c}), paramno, modelfile{iFn}); return;
-    end;
-  end;
-  % transform into row vector and right pad with zeros --
-  conmat=zeros(numel(connames), paramno);
-  for c=1:numel(convec)
-    conmat(c,1:numel(convec{c}))=convec{c};
-  end;
-
-  % calculate contrasts --
+        numel(convec{c}), paramno, modelfile{iFn});
+      return
+    end
+  end
+  % 3.6 transform into row vector and right pad with zeros --
+  conmat = zeros(numel(connames), paramno);
+  for c = 1:numel(convec)
+    conmat(c,1:numel(convec{c})) = convec{c};
+  end
```

Comment:
Tip: If all cells of `convec `are row-vector, I think `conmat = cat(1,convec{:});` is similar to your for loop

Reply:
This is not written by me. I just checked and I think they do have similar features. I have updated the code based on your suggestion. Now sure if we need to ask Dominik to check as he is the author of this line of code.